pkgs/sketch_pad/lib/docs.dart

```diff
@@ -0,0 +1,123 @@
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+
+    return Container(
+      decoration: BoxDecoration(
+        color: theme.scaffoldBackgroundColor,
+        border: Border(
+          top: Divider.createBorderSide(
+            context,
+            width: 8.0,
+            color: theme.colorScheme.surface,
+          ),
+        ),
+      ),
+      padding: const EdgeInsets.all(denseSpacing),
+      child: Stack(
+        children: [
+          ValueListenableBuilder(
+            valueListenable: widget.appModel.currentDocs,
+            builder: (context, DocumentResponse? docs, _) {
+              // TODO: Consider showing propagatedType if not null.
+
+              var title = _cleanUpTitle(docs?.elementDescription);
+              if (docs?.deprecated == true) {
+                title = '$title (deprecated)';
+              }
+
+              // TODO: should the markdown text be selectable?
+              return Column(
+                crossAxisAlignment: CrossAxisAlignment.stretch,
+                children: [
+                  Text(
+                    title,
+                    style: theme.textTheme.titleMedium,
+                    maxLines: 1,
+                    overflow: TextOverflow.ellipsis,
+                  ),
+                  const SizedBox(height: denseSpacing),
+                  Expanded(
+                    child: Markdown(
+                      data: docs?.dartdoc ?? '',
+                      padding: const EdgeInsets.only(left: denseSpacing),
+                      onTapLink: _handleMarkdownTap,
+                    ),
+                  ),
+                ],
+              );
+            },
+          ),
+          Padding(
+            padding: const EdgeInsets.all(denseSpacing),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.end,
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                MiniIconButton(
+                  icon: Icons.close,
+                  tooltip: 'Close',
+                  onPressed: _closePanel,
+                  small: true,
+                )
+              ],
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  void _closePanel() {
+    widget.appModel.docsShowing.value = false;
+  }
+
+  String _cleanUpTitle(String? title) {
+    if (title == null) return '';
+
+    // "(new) Text(\n  String data, {\n  Key? key,\n  ... selectionColor,\n})"
+
+    // Remove ws right after method args.
+    title = title.replaceAll('(\n  ', '(');
+
+    // Remove ws before named args.
+    title = title.replaceAll('{\n  ', '{');
+
+    // Remove ws after named args.
+    title = title.replaceAll(',\n}', '}');
+
+    return title.replaceAll('\n', '').replaceAll('  ', ' ');
+  }
+
+  void _handleMarkdownTap(String text, String? href, String title) {
+    if (href != null) {
+      url_launcher.launchUrl(Uri.parse(href));
```

Comment:
Since this can come from packages or user's own API docs, consider using `Uri.tryParse` here.

Reply:
+1 to being defensive gere - I'll switch to `tryParse`.